Assignable Keyboard Shortcuts Updates

WebShells opened the pull request:
Few updates regarding HotKeys/ Keyboard Shortcut section.

. Added Shortcut TAB under config section where user can re-assign the shortcuts as needed
. Shortcuts Help Modal in Sales Register now reflects the assigned shortcuts
. Updates JS section in register in order to retreive assigned keys from database.

To be fixed:
. Migration file
. Success message on submit

WebShells committed:
Keyboard Shortcuts Updates

Few updates regarding HotKeys/ Keyboard Shortcut section.

. Added Shortcut TAB under config section where user can re-assign the shortcuts as needed
. Shortcuts Help in Sales Register now reflects the assigned shortcuts
. Updates JS section in order to retreive assigned keys from database.

Still need to submit migration file and few fixes.

WebShells committed:
Removing Lang Duplicates

WebShells committed:
Code alignment

jekkos commented on application/libraries/Sale_lib.php:
```diff
@@ -56,7 +56,23 @@ public function get_invoice_type_options()
+	public function get_key_shortcuts_options()
+	{
+		$key_shortcuts = array();
+		$key_shortcuts['27 | ESC'] = $this->CI->lang->line('sales_key_1');
```
this can be done in a loop

jekkos commented:
@WebShells we might want to freeze master branch until the CI4 upgrade is done, merging this now will only create more conflicts later on. Any new functionality should be built on top of CI4.

WebShells commented:
> @WebShells we might want to freeze master branch until the CI4 upgrade is done, merging this now will only create more conflicts later on. Any new functionality should be built on top of CI4.

@jekkos do you want me to apply the changes to fit to CI4 upgrade and submit the PR to that branch ? or shall i wait till the upgrade is done ?

objecttothis commented:
> > @WebShells we might want to freeze master branch until the CI4 upgrade is done, merging this now will only create more conflicts later on. Any new functionality should be built on top of CI4.
> 
> @jekkos do you want me to apply the changes to fit to CI4 upgrade and submit the PR to that branch ? or shall i wait till the upgrade is done ? 

I recommend we still set this PR to merge with the master but write the changes to work with CI4. Then it will be ready to test/merge after the CI4 branch has been merged into the master.

I agree that the master branch needs to be frozen.

jekkos commented on application/libraries/Sale_lib.php:
```diff
@@ -56,7 +56,23 @@ public function get_invoice_type_options()
+	
+	public function get_key_shortcuts_options()
+	{
+		$key_shortcuts = array();
```
What could be cleaner here is to approach this in OOP fashion and have a proper `Shortcut` class or a central dict, that has a couple of fields like `keycode`, `lang_line`. Then most of the repetitive parts in this code can be generated dynamically, by filling in the correct variable at every step

jekkos commented on application/views/sales/help.php:
```diff
@@ -8,50 +8,51 @@
 				<tr>
-					<td><code>ESC</code></td>
+					
 					<td><?php echo $this->lang->line('sales_key_cancel'); ?></td>
```
here you would have one loop going over the datastructure and printing the rows in generic fashion

jekkos commented:
> > > @WebShells we might want to freeze master branch until the CI4 upgrade is done, merging this now will only create more conflicts later on. Any new functionality should be built on top of CI4.
> > 
> > 
> > @jekkos do you want me to apply the changes to fit to CI4 upgrade and submit the PR to that branch ? or shall i wait till the upgrade is done ?
> 
> I recommend we still set this PR to merge with the master but write the changes to work with CI4. Then it will be ready to test/merge after the CI4 branch has been merged into the master.
> 
> I agree that the master branch needs to be frozen.

Fine by me, I guess this won't be a big one. But we need a migration at least otherwise there might be issues in the register @WebShells

WebShells committed:
Migration Script

Added Migration script in order to insert default shortcut keys values into db.

WebShells commented:
> Fine by me, I guess this won't be a big one. But we need a migration at least otherwise there might be issues in the register @WebShells

Added.

objecttothis reviewed (changes requested):
Super.  Once #3592 is merged with the master you'll need to modify these changes to CI4 formatting and then we can approve and merge.  If you want to get a head start, you can take a look at these files in the ci4-upgrade branch and change it now.

WebShells committed:
Renamed Lang File

Renamed ur_PK to ur-PK as listed in locale_helper.php

objecttothis commented:
@WebShells now that CI4 has been merged into the master can you rebase these changes?

WebShells commented:
> @WebShells do you want to bring this branch up to date with the master and take a look at the review comments so we can merge this PR?

Ill handle that @objecttothis
